cpp/inline/util/EigenUtil.inl

```diff
@@ -213,6 +213,13 @@ auto sigmoid(const Array& array) {
   return 1.0 / (1.0 + (-array).exp());
 }
 
+template <typename Array>
+inline int argmax(const Array& arr) {
+  Eigen::Index maxIndex;
+  arr.maxCoeff(&maxIndex);
+  return static_cast<int>(maxIndex);
```

Comment:
Do you require `Array` to be 1D? What happens here is you pass a multi-dimensional array?